testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -1107,6 +1092,19 @@
         )
 end
 
+# In case of a search reindex not having finished yet, keep retrying until successful search or timeout
+When(/^I click on the search button$/) do
+  click_button_and_wait('Search', match: :first)
+  # after a search reindex, the UI will show a "Could not connect to search server" followed by a false "No matches found" for a while
+  if has_text?('Could not connect to search server.', wait: 0)
+    repeat_until_timeout(message: 'Could not perform a successful search after reindexation', timeout: 10) do
+      break unless has_text?('Could not connect to search server.', wait: 0) || has_text?('No matches found', wait: 0)
+      sleep 1
+      click_button('Search', match: :first, wait: false)
+    end
+  end
```

Comment:
```suggestion
# after a search reindex, the UI will show a "Could not connect to search server" followed by a false "No matches found" for a while
  repeat_until_timeout(message: 'Could not perform a successful search after reindexation', timeout: 10) do
    click_button('Search', match: :first)
    break unless has_text?('Could not connect to search server.', wait: 1) || has_text?('No matches found', wait: 1)
  end
```

Reply:
That would indeed be much simpler, and that was also my first idea.

That will not work though because of the weird behaviour explained in the comment:
```
# after a search reindex, the UI will show a "Could not connect to search server" followed by a false "No matches found" for a while
```
For more information, ask Vanda.

Reply:
Also, to add, the sleep 1 actually helps us not overload the system and keep retrying every single possible moment - gives the testsuite some time to "breathe" before retrying, without taking up too much time - it takes about as long as my previous version, I tested it, but without constantly overloading the system. Kind of like awaitility works for java, where you can set a retry interval - this sleep immitates that behaviour

Reply:
I just realised you added the wait to waiting for the text but that is _not_ the same thing - it's not about the text showing up, it's about us waiting _before_ retrying the search again

Reply:
Ok :)